.github/workflows/test-all-warehouses.yml

```
@@ -45,3 +45,17 @@ jobs:
       dbt-version: ${{ matrix.dbt-version }}
       generate-data: ${{ inputs.generate-data || false }}
     secrets: inherit
+
+  notify_failures:
+    name: Notify Slack
+    needs: test
+    if: |
+      always() &&
+      contains(needs.test.result, 'failure') &&
```

Is there a reason not to do string comparison?

```suggestion
      needs.test.result == 'failure' &&
```

---

Actually not sure it'd work, I thinkkk there's a reason why this is how its mentioned to do in the docs 
Anyway would rather leave it like that instead of taking the chance, it works :)